support package that contains '-' in its name, like syslog-ng
http://reductivelabs.com/trac/puppet/ticket/628
--- lib/puppet/provider/package/ports.rb.orig       Wed May  9 17:12:39 2007
+++ lib/puppet/provider/package/ports.rb    Wed May  9 17:13:57 2007
@@ -47,7 +47,7 @@
         match = $2
         info = $3

-        unless pkgstuff =~ /^(\w+)-([0-9].+)$/
+        unless pkgstuff =~ /^(\S+)-([^-\s]+)$/
             raise Puppet::PackageError,
                 "Could not match package info '%s'" % pkgstuff
         end
